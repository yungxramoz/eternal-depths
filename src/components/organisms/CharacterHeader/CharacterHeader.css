.character-information {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  gap: 1em;
  padding: 1em 2em 8em 2em;
  width: 100%;
}

.character-stats {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 1em;
  flex-grow: 1;
}
