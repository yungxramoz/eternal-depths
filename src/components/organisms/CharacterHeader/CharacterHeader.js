import React from 'react'
import CharacterAvatar from '../../molecules/CharacterAvatar/CharacterAvatar'
import HpLightProgressBar from '../../molecules/ProgressBar/HpLightProgressBar'
import XpLightProgressBar from '../../molecules/ProgressBar/XpLightProgressBar'
import './CharacterHeader.css'

const CharacterHeader = ({ character, maxHp, maxXp }) => {
  return (
    <div className="character-information">
      <CharacterAvatar look={character.look} />
      <div className="character-stats">
        <HpLightProgressBar currentHp={character.hp} maxHp={maxHp} />
        <XpLightProgressBar
          currentLvl={character.level}
          currentXp={character.xp}
          maxXp={maxXp}
        />
      </div>
    </div>
  )
}

export default CharacterHeader
