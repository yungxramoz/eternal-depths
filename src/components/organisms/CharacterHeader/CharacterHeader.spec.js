import { render } from '@testing-library/react'
import React from 'react'
import CharacterHeader from './CharacterHeader'

jest.mock('../../molecules/CharacterAvatar/CharacterAvatar', () => {
  return ({ look }) => <div>CharacterAvatar, look: {look}</div>
})

describe('CharacterHeader', () => {
  const character = {
    look: 'test-look',
    hp: 50,
    level: 1,
    xp: 10,
  }
  const maxHp = 100
  const maxXp = 20

  it('renders correctly', () => {
    render(
      <CharacterHeader character={character} maxHp={maxHp} maxXp={maxXp} />,
    )
    expect(document.body).toMatchSnapshot()
  })
})
