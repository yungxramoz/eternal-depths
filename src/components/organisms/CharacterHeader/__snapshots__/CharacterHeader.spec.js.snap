// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterHeader renders correctly 1`] = `
<body>
  <div>
    <div
      class="character-information"
    >
      <div>
        CharacterAvatar, look: 
        test-look
      </div>
      <div
        class="character-stats"
      >
        <div
          class="light-progress-container"
        >
          <p
            class="light-hint-text"
          >
            HP (
            50
            /
            100
            )
          </p>
          <div
            class="light-progress-bar"
          >
            <div
              class="rpgui-progress-track light-progress-track"
            >
              <div
                class="rpgui-progress-fill red progress-animation"
                style="left: 0px; width: 50%;"
              />
            </div>
          </div>
        </div>
        <div
          class="light-progress-container"
        >
          <p
            class="light-hint-text"
          >
            Level 
            1
             (
            10
            /
            20
            )
          </p>
          <div
            class="light-progress-bar"
          >
            <div
              class="rpgui-progress-track light-progress-track"
            >
              <div
                class="rpgui-progress-fill blue progress-animation"
                style="left: 0px; width: 50%;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
